tests/infrastructure/instance_types/test_common_vm_preference.py

```diff
@@ -59,31 +61,50 @@
     "rhel.8.dpdk",
     "rhel.9.dpdk",
 ]
+GENERIC_FIRMWARE_PREFERENCE_LIST = [
+    "linux",
+    "linux.efi",
+    "legacy",
+]
 
 
-def start_vm_with_cluster_preference(client, preference_name, namespace_name):
-    cluster_preference = VirtualMachineClusterPreference(name=preference_name)
-    cluster_spec = cluster_preference.instance.spec
-    cpu_guest = cluster_spec.requirements.cpu.guest
-    spread_options = getattr(cluster_spec.cpu, "spreadOptions", None)
+def _extract_resources_from_cluster_spec(cluster_spec):
+    def _get_nested_attribute(obj, path):
+        for attr in path.split("."):
+            obj = getattr(obj, attr, None)
+        return obj
 
-    sockets = cpu_guest
+    cpu_guest = _get_nested_attribute(obj=cluster_spec, path="requirements.cpu.guest")
+    memory_guest = (
+        _get_nested_attribute(obj=cluster_spec, path="requirements.memory.guest") or Images.Rhel.DEFAULT_MEMORY_SIZE
+    )
+    spread_options = _get_nested_attribute(obj=cluster_spec, path="cpu.spreadOptions")
```

Comment:
you have `VirtualMachineClusterPreference` object, get what is needed by accessing the resource.

Reply:
requirements.cpu.guest , requirements.memory.guest , cpu.spreadOptions are all optional and i feel it's better to check if they are present.

Reply:
this code is much more complex than using `.get`